app/controllers/admin/schools_controller.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Admin
+  class SchoolsController < Admin::BaseController
+    skip_after_action :verify_authorized, only: :index
+
+    def index
+      @query = params[:query]
+      @schools = policy_scope(School).order(:name).page(params[:page]).per(20)
+      if @query.present?
+        @schools = @schools.search_by_name_or_urn(@query)
```

Comment:
Can we include local_authority and induction_coordinators when fetching?

Reply:
We're still N+1 on local authority, because of the way we fetch it. Open to suggestions

Reply:
Can you do something like
```
@schools = @schools.search_by_name_or_urn(@query).joins(:local_authorities).includes(:local_authorities)
```
Is Rails clever enough to work the `through` relation into its query?

Reply:
Nope, since we call `local_authority` on School, which is 
```
def local_authority
    school_local_authorities.latest.first&.local_authority
  end
```

Reply:
ufff

Yeah, I don't think we will find a way to get rid of n+1s without storing some more things in the database :/ Like `current_local_authority` on a school model.

Reply:
This is an admin thing, and there are only a few of them. We also paginate to 20. Are you happy to leave this as-is for now?

Reply:
Yes, I think the level of annoyance to avoid is it high enough to keep it for now and pick it up again when we know for sure that it's a problem worth fixing.